tests/kernels/moe/test_pplx_cutlass_moe.py

```diff
@@ -0,0 +1,362 @@
+# SPDX-License-Identifier: Apache-2.0
+import dataclasses
+import traceback
+from typing import Callable
+
+import pytest
+import torch
+
+from vllm import _custom_ops as ops
+from vllm.config import VllmConfig, set_current_vllm_config
+from vllm.model_executor.layers.activation import SiluAndMul
+from vllm.model_executor.layers.fused_moe.cutlass_moe import CutlassExpertsFp8
+from vllm.model_executor.layers.fused_moe.fused_moe import fused_topk
+from vllm.model_executor.layers.fused_moe.modular_kernel import (
+    FusedMoEModularKernel)
+from vllm.model_executor.layers.fused_moe.pplx_prepare_finalize import (
+    PplxPrepareAndFinalize)
+from vllm.platforms import current_platform
+
+try:
+    from pplx_kernels import AllToAll  # or AllToAllInternode?
+    from pplx_kernels.nvshmem import (nvshmem_alloc_empty_unique_id,
+                                      nvshmem_finalize, nvshmem_get_unique_id,
+                                      nvshmem_init)
+    has_pplx = False
+except ImportError:
+    has_pplx = False
+
+from torch.multiprocessing import (
+    spawn)  # pyright: ignore[reportPrivateImportUsage]
+from typing_extensions import Concatenate, ParamSpec
+
+NUM_EXPERTS = [40, 64]
+TOP_KS = [6, 8]
+
+P = ParamSpec("P")
```

Comment:
We should probably put all these multiprocess utilities in a separate file now since they are also used by test_pplx_moe.py